modules/nextflow/src/main/groovy/nextflow/cli/CmdAuth.groovy

```diff
@@ -0,0 +1,1324 @@
+            result << '  2. Wait for user to complete authentication in web browser'
+            result << '  3. Generate and save access token to home-directory Nextflow config'
+            result << '  4. Configure tower.accessToken, tower.endpoint, and tower.enabled settings'
+            result << ''
+        }
+    }
+
+    class LogoutCmd implements SubCmd {
+
+        @Override
+        String getName() { 'logout' }
+
+        @Override
+        void apply(List<String> args) {
+            validateArgumentCount(args, name)
+
+            // Check if TOWER_ACCESS_TOKEN environment variable is set
+            def envToken = System.getenv('TOWER_ACCESS_TOKEN')
+            if (envToken) {
+                println ""
+                ColorUtil.printColored("WARNING: TOWER_ACCESS_TOKEN environment variable is set.", "yellow bold")
+                println " ${ColorUtil.colorize('nextflow auth logout', 'dim cyan')}${ColorUtil.colorize(' only removes credentials from Nextflow config files.', 'dim')}"
+                ColorUtil.printColored(" The environment variable will remain unaffected.", "dim")
+                println ""
+            }
+
+            // Check if tower.accessToken is set
+            def config = readConfig()
+            def existingToken = config['tower.accessToken']
+            def endpoint = config['tower.endpoint'] ?: 'https://api.cloud.seqera.io'
+
+            if (!existingToken) {
+                ColorUtil.printColored("Error: No authentication token found in Nextflow config.", "red")
+                println "Config file: ${ColorUtil.colorize(getConfigFile().toString(), 'magenta')}"
+                return
+            }
+
+            ColorUtil.printColored(" - Found authentication token in config file: ${ColorUtil.colorize(getConfigFile().toString(), 'magenta')}", "dim")
+
+            // Prompt user for API URL if not already configured
+            def apiUrl = endpoint as String
+            if (!apiUrl || apiUrl.isEmpty()) {
+                apiUrl = promptForApiUrl()
+            } else {
+                ColorUtil.printColored(" - Using Seqera Platform endpoint: ${ColorUtil.colorize(apiUrl, 'magenta')}", "dim")
+            }
+
+            // Validate token by calling /user-info API
+            try {
+                def userInfo = callUserInfoApi(existingToken as String, apiUrl)
+                ColorUtil.printColored(" - Token is valid for user: ${ColorUtil.colorize(userInfo.userName as String, 'cyan bold')}", "dim")
+
+                // Only delete PAT from platform if this is a cloud endpoint
+                if (isCloudEndpoint(apiUrl)) {
+                    def tokenId = decodeTokenId(existingToken as String)
+                    deleteTokenViaApi(existingToken as String, apiUrl, tokenId)
+                } else {
+                    println " - Enterprise installation detected - PAT will not be deleted from platform."
+                }
+
+                removeAuthFromConfig()
+
```

Comment:
What happens if there was a previous token already defined in the config. Could it be deleted by mistake?